### PR TITLE
Add support for is_digit and regular expression difference

### PR DESCRIPTION
**Pull request by ajreynol**
Towards support for the strings standard.  This adds support for str.is_digit and re.diff, which both can be eliminated eager during preprocessing.

### CODE REVIEW

File: src/api/cvc4cpp.cpp [hunk 1]
```diff
@@ -263,13 +263,15 @@ const static std::unordered_map<Kind, CVC4::Kind, KindHashFunction> s_kinds{
     {STRING_LEQ, CVC4::Kind::STRING_LEQ},
     {STRING_PREFIX, CVC4::Kind::STRING_PREFIX},
     {STRING_SUFFIX, CVC4::Kind::STRING_SUFFIX},
+    {STRING_ISDIGIT, CVC4::Kind::STRING_ISDIGIT},
```
Comment (4tXJ7f) [inline src/api/cvc4cpp.cpp]: Wouldn't it be more consistent to call this kind `STRING_IS_DIGIT` (to match `STRING_TO_REGEXP`/`STRNG_IN_REGEXP`?

Comment (ajreynol) [reply src/api/cvc4cpp.cpp]: Sure, ill update

Comment (ajreynol) [reply src/api/cvc4cpp.cpp]: We should really consider changing STRING_STRREPL -> STRING_REPLACE too then :)

Comment (4tXJ7f) [reply src/api/cvc4cpp.cpp]: I agree, we should make everything consistent now that the standard exists (also names like `STRING_CONTAINS`, `STRING_FROM_INT`, `STRING_TO_INT`).

Comment (ajreynol) [reply src/api/cvc4cpp.cpp]: Let's do this on a separate PR.

**Pull request merged**